packages/edit-site/src/components/sidebar-dataviews/default-views.js

```diff
@@ -52,7 +53,16 @@ export const DEFAULT_VIEWS = {
 			title: __( 'All pages' ),
 			slug: 'all',
 			icon: pages,
-			view: DEFAULT_PAGE_BASE,
+			view: {
+				...DEFAULT_PAGE_BASE,
+				filters: [
+					{
+						field: 'status',
+						operator: OPERATOR_IS_NONE,
+						value: 'trash',
+					},
+				],
```

Comment:
Although this filter is not needed for the query which is run to fetch all the pages, because `trash` is excluded by default, but this needed in the JS filter we are doing here - https://github.com/WordPress/gutenberg/pull/62028/files#diff-67163f722d76f852faaff5a7c0c6046d411846fc118479e62b70273363272d5cR27

Reply:
Why are we doing JS filtering? IMO, we shouldn't because the REST API can be altered and extended in multiple ways so the only way to get a reliable result is to use the same REST API used in the dataviews.

Reply:
I need to spend more time with data views, so I can't respond to the above point, but I don't think we need this filter here at all. The resulting filter badge, when clicked, does nothing.